### ISSUE TITLE
FakeContext does not implement IAsyncQueryProvider

### ISSUE THREAD

**Issue by slee-accesso** [ReversePOCO/EntityFramework-Reverse-POCO-Code-First-Generator#674]
Hi Simon,

You addressed some of the async issues in a previous post. However, I'm now seeing downstream failures that I missed in earlier testing - mainly I didn't test with an actual query - I only was looking for FirstOrDefaultAsync against the data set directly. Now when executing with an actual query, I get an exception.

Here's my test:
```csharp
[Fact]
public async void FakeContext_QuerySucceeds()
{
    var fakeContext = new FakeContext() {WwSales = {new WwSale() {SaleNo = 123}}};

    // Directly against the dataset, no query - both work
    var result1 = fakeContext.WwSales.FirstOrDefault();            // succeeds
    Assert.Equal(123, result1.SaleNo);

    var result2 = await fakeContext.WwSales.FirstOrDefaultAsync(); // succeeds
    Assert.Equal(123, result2.SaleNo);

    // with query - only synchronous works
    var query = from w in fakeContext.WwSales select new {QueriedSaleNo = w.SaleNo};
    
    var result3 = query.FirstOrDefault();            // succeeds
    Assert.Equal(123, result3.QueriedSaleNo);
    
    var result4 = await query.FirstOrDefaultAsync(); // dies a horrible death
    Assert.Equal(123, result4.QueriedSaleNo);
}
```
and the exception:
```
The provider for the source 'IQueryable' doesn't implement 'IAsyncQueryProvider'.
Only providers that implement 'IAsyncQueryProvider' can be used for Entity Framework asynchronous operations.
```

**Comment by sjh37** [comment]
Hi @slee-accesso 
I've made some recent changes in changeset https://github.com/sjh37/EntityFramework-Reverse-POCO-Code-First-Generator/commit/f0a85890d2e37f3b63ab9601439daf33c0858c94

Update your EF.Reverse.POCO.v3.ttinclude file to include those changes, or take the latest version from here:
https://raw.githubusercontent.com/sjh37/EntityFramework-Reverse-POCO-Code-First-Generator/master/EntityFramework.Reverse.POCO.Generator/EF.Reverse.POCO.v3.ttinclude

Please let me know if this solves the issue you are having.

**Comment by slee-accesso** [comment]
Hi Simon,

Thanks. I implemented those changes 6 days ago (I was the one that brought up the async issues last week). So - those changes helped the FirstOrDefaultAsync() against a data set directly - but not with a query. In my tests above, your fix last week made test number 2 work, which previously failed - but test 4 still fails.

 - Sean

**Comment by sjh37** [comment]
This has now been released in v3.4.0
